#!/usr/bin/env python

PACKAGE = "depth_processing"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#Filters
gen.add("voxel_leaf_size", double_t, 0, "The leaf size for voxel filter", 0.01, 0.001, 0.1)
gen.add("mean_k", int_t, 0, "The number of nearest neighbors to use for SOR filter", 50, 1, 100)
gen.add("std_dev", double_t, 0, "The standard deviation multiplier for SOR filter", 1.0, 0.1, 5.0)
gen.add("passthrough_limit_min", double_t, 0, "Minimum limit for passthrough filter", 0.0, 0.0, 2.0)
gen.add("passthrough_limit_max", double_t, 0, "Maximum limit for passthrough filter", 1.239, 0.5, 5.0)

#ICP
gen.add("max_iterations", int_t, 0, "Maximum Iterations", 50, 1, 200)
gen.add("max_correspondence_distance", double_t, 0, "Max Correspondence Distance", 0.05, 0.001, 1.0)
gen.add("transformation_epsilon", double_t, 0, "Transformation Epsilon", 1e-8, 1e-10, 1e-5)
gen.add("euclidean_fitness_epsilon", double_t, 0, "Euclidean Fitness Epsilon", 1e-8, 1e-10, 1e-5)
gen.add("ransac_outlier_rejection_threshold", double_t, 0, "RANSAC Outlier Rejection Threshold", 0.05, 0.001, 1.0)
gen.add("use_reciprocal_correspondences", bool_t, 0, "Use Reciprocal Correspondences", False)


exit(gen.generate(PACKAGE, "depth_processing", "Filters"))
